Assets/MixedRealityToolkit/Common/Shaders/Standard.shader

```diff
@@ -456,46 +491,61 @@ Shader "MixedRealityToolkit/Standard"
                 float planeDistance = PointVsPlane(i.worldPosition.xyz, _ClipPlane);
 #if defined(_CLIPPING_PLANE_BORDER)
                 fixed3 planeBorderColor = lerp(_ClippingPlaneBorderColor, fixed3(0.0, 0.0, 0.0), planeDistance / _ClippingPlaneBorderWidth);
-                albedo.rgb += step(planeDistance, _ClippingPlaneBorderWidth) * planeBorderColor;
+                albedo.rgb += planeBorderColor * ((planeDistance < _ClippingPlaneBorderWidth) ? 1.0 : 0.0);
 #endif
 #if defined(_ALPHA_CLIP)
-                albedo *= step(0.0, planeDistance);
+                albedo *= (planeDistance > 0.0);
 #else
                 albedo *= saturate(planeDistance);
 #endif
 #endif
 
-#if defined(_ROUND_CORNERS) || defined(_BORDER_LIGHT) || defined(_INNER_GLOW)
+#if defined(_DISTANCE_TO_EDGE)
                 fixed2 distanceToEdge;
                 distanceToEdge.x = abs(i.uv.x - 0.5) * 2.0;
                 distanceToEdge.y = abs(i.uv.y - 0.5) * 2.0;
 #endif
 
                 // Rounded corner clipping.
 #if defined(_ROUND_CORNERS)
-                fixed cornerCircleRadius = (_RoundCornerRadius - _RoundCornerMargin) * i.scale.z;
-                fixed2 roundCornerPosition = distanceToEdge * 0.5 * i.scale.xy;
-                fixed2 cornerCircleDistance = (i.scale.xy * 0.5) - cornerCircleRadius - _RoundCornerMargin * i.scale.xy;
+                fixed2 halfScale = i.scale.xy * 0.5;
+                fixed2 roundCornerPosition = distanceToEdge * halfScale;
+
+                fixed cornerCircleRadius = saturate(_RoundCornerRadius - _RoundCornerMargin) * i.scale.z;
+                fixed2 cornerCircleDistance = halfScale - (_RoundCornerMargin * i.scale.z) - cornerCircleRadius;
+
                 fixed roundCornerClip = RoundCorners(roundCornerPosition, cornerCircleDistance, cornerCircleRadius);
 #endif
 
                 albedo *= _Color;
 
                 // Hover light.
 #if defined(_HOVER_LIGHT)
-                fixed pointToHover = (1.0 - saturate(length(_HoverPosition - i.worldPosition.xyz) / _HoverRadius)) * _HoverColor.a;
+                fixed pointToHover0 = HoverLight(_HoverLight0, i.worldPosition.xyz, _HoverLightColor0.a);
+#if defined(_MULTI_HOVER_LIGHT)
+                fixed pointToHover1 = HoverLight(_HoverLight1, i.worldPosition.xyz, _HoverLightColor1.a);
```

Comment:
Oh I see, we need to define a fixed point for each in the shader itself.

Reply:
Is there a way to make this more generic to support any number of HoverLights?

Reply:
There isn't an explicit hard limit, but I wanted to settle on a number that didn't incur much overhead (especially for the HoloLens and "Core" Windows MR PCs). Two seemed like a good number since usually we have zero to two motion controllers/pointers (I'm sure that will change before we know it haha).

If there is an outcry for more hover lights, I'm happy to add a more generic solution in the future!

Reply:
Could we do 3 for now?

Cause chances are you'll have gaze as well as two controller pointers.

Reply:
3 lights will be an easy change, I'll make those changes and upload them tomorrow. Since you are  most likely to need 2+ hover lights on an immersive HMD we should have performance to spare. :)